alf/environments/igibson/suite_igibson.py

```diff
@@ -0,0 +1,384 @@
+        when called by subprocesses. An explanation of the non-contiguous memory caused
+        by numpy tranpose can be found in the following:
+
+        https://stackoverflow.com/questions/26998223/
+        """
+
+        if flatten:
+            np_array = np_array.flatten()
+        return np_array.copy()
+
+
+@alf.configurable()
+class StackRGBDWrapper(gym.ObservationWrapper):
+    """This wrapper stacks rgb and depth observation field together into an array of shape (H, W, 4)
+
+    Note that the observation space must include `rgb` and `depth`, otherwise an error will be raised.
+    """
+
+    def __init__(self, env):
+        super().__init__(env)
+        observation_space = env.observation_space
+        self.observation_space = self.transform_space(observation_space)
+
+    def transform_space(self, observation_space):
+        assert isinstance(observation_space, gym.spaces.dict.Dict)
+        rgb_space = observation_space.spaces['rgb']
+        depth_space = observation_space.spaces['depth']
+        try:
+            assert not np.isscalar(rgb_space.low) and not np.isscalar(
+                rgb_space.low)
+            assert not np.isscalar(depth_space.low) and not np.isscalar(
+                depth_space.low)
+        except AssertionError as e:
+            print(e)
+
+        rgbd_low = np.vstack((rgb_space.low, depth_space.low))
+        rgbd_high = np.vstack((rgb_space.high, depth_space.high))
+        assert rgbd_low.shape == rgbd_high.shape
+        rgbd_space = gym.spaces.Box(
+            low=rgbd_low, high=rgbd_high, dtype=np.uint8)
+
+        observation_space.spaces.pop('rgb')
+        observation_space.spaces.pop('depth')
+        observation_space.spaces['rgbd'] = rgbd_space
+        return observation_space
+
+    def observation(self, observation):
+        return self.transform_observation(observation)
+
+    def transform_observation(self, observation):
+        rgbd_obs = np.vstack((observation['rgb'], observation['depth']))
+        observation.pop('rgb')
+        observation.pop('depth')
+        observation['rgbd'] = rgbd_obs
+        return observation
+
+
+@alf.configurable()
+class OneHotIntWrapper(BaseObservationWrapper):
+    """Transforms the data type of one-hot input from float to int"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        obs_low = np.full_like(
+            observation_space.low, fill_value=0, dtype=np.uint8)
+        obs_high = np.full_like(
+            observation_space.high, fill_value=255, dtype=np.uint8)
+        assert obs_low.shape == obs_high.shape
+        observation_space = gym.spaces.Box(
+            low=obs_low, high=obs_high, dtype=np.uint8)
+        return observation_space
+
+    def transform_observation(self, observation):
+        return (observation * 255).astype(np.uint8)
+
+
+@alf.configurable()
+class ImageIntWrapper(BaseObservationWrapper):
```

Comment:
Probably we should set the image type to uint8 in the first place when it's generated by the simulator, instead of using a wrapper to do this (which is inefficient).